param/__init__.py

```diff
@@ -314,6 +314,28 @@ def _get_min_max_value(min, max, value=None, step=None):
     return min, max, value
 
 
+def _deserialize_from_path(ext_to_routine, path, type_name):
+    """Call deserialization routine with path according to extension"""
```

Comment:
@jbednar I suppose there's some ambiguity in the title. Deserialization could refer to what Pandas is doing, i.e. interpreting the contents of a file as a data frame. However, the deserialization this PR refers to is the deserialization `param` is doing, i.e. interpreting the contents of a JSON file as a Parameterized instance. Pandas' deserialization from file only occurs when the DataFrame parameter is assigned a string in the JSON file.

I think it could make sense to elevate this functionality so that any time a `param.DataFrame` parameter is passed a string, it treats it as a file path and tries to read it. I can't think of any reason off the top of my head why you shouldn't, nor why the functionality shouldn't be extended to `param.Array` parameters. The additional functionality could open up new questions which are a lot more complicated, particularly w.r.t. any persistent relationship to the file. For example, should a DataFrame initially read from a file be updated if the `param.DataFrame` instance is mutated? If one serializes a parameterized instance into JSON, then, does the file name get serialized as well? The answer to both may be "no," but I remember having discussions with @jbednar and I believe @jlstevens in a similar vein a few years back when I made this PR.

In any event, I believe I delivered on the functionality agreed upon and resolved all issues circa 2021. Though I'm happy to make small changes to conform to style, etc., I'd rather not be involved in developing more functionality at this time.

Reply:
Ah, apologies. It looks as though @jbednar did ask about this in 2021. I missed it; that's my fault. If there's a consensus that this is what's wanted, I can spend some time revamping this functionality. Is it what's wanted?

Reply:
I think there are lots of extra features that could be implemented around this but I also think that not all these extra features are necessarily desirable or worth the extra code complexity. My feeling is that the basic idea behind this PR makes sense and that the job then is to clearly document the known limitations. No need to let perfect be the enemy of good! (and issues such as detecting whether a dataframe has been mutated are tricky enough problems in their own right)